support vultr api v2 and python3

gricn opened the pull request.

gricn committed:
support vultr api v2 and python3

alexzorin commented on vultr-dns.py:
```diff
@@ -1,43 +1,49 @@
+def vultr_request(method, path, data=None, json=None):
+    url = "https://api.vultr.com/v2{}".format(path)
 
-    resp = requests.request(method, url, data=data, headers={
-                            "API-Key": VULTR_API_KEY})
+    resp = requests.request(
+        method,
+        url,
+        data=data,
+        json=json,
+        headers={"Authorization": "Bearer " + VULTR_API_KEY},
+    )
     resp.raise_for_status()
-    if resp.headers['Content-Type'] == 'application/json':
+    if resp.headers["Content-Type"] == "application/json":
         return resp.json()
     return resp.text
```
`data` and `json` appear to be redundant. Let's keep only one of them.

gricn replied:
Okay, so let's just leave the `json`. As [requests doc](https://docs.python-requests.org/en/master/user/quickstart/#more-complicated-post-requests) says, using json parameter will change the `Content-Type` in the header to `application/json`. That's Vultr API v2 needs.

alexzorin commented on vultr-dns.py:
```diff
@@ -46,55 +52,75 @@ def find_zone_for_name(domain):
+        "type": "TXT",
+        "data": '"{}"'.format(txt_value),
+    }
+    print("/domains/{}/records".format(zone))
```
Leftover debugging?

```suggestion
```

gricn replied:
Aha, that‘s my mistake.

alexzorin commented on vultr-dns.py:
```diff
@@ -1,43 +1,49 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
```
Supporting Python 3 is one thing (like with the `print` fix, thanks!), but requiring it is another.

Python 2.7 is not quite dead; it's still the default Python in LTS distributions like Ubuntu Focal.

Is there a particular reason you had in mind for this change?

gricn replied:
Before PR, I have noticed it. There is only a `master` branch in this repository, however, I have no choice but to merge into it. 🤣 🤣 

Maybe we could make a new branch like `Python3` and update the `README` file?

BTW, since I am a newbie here, what should I do next? Creating a new PR after a new branch init?

alexzorin replied:
Well, with your changes, the program works in either Python 2 or 3, right?

So if we just leave `#!/usr/bin/env python` as it is, the program will use the default Python interpreter on the system, which may be either Python 2 or Python 3 (it depends on the system).

To me, that feels like the right thing to do. I don't see any reason to maintain a separate branch.

gricn replied:
I have tested in my server, both python2 and 3 works fine. Just do as you say 😄 

Update `README` to tell users to change `#!/usr/bin/env python` according to their system's setting. Eg. I have to use `#!/usr/bin/env python3` to run this script by python3 and `#!/usr/bin/env python2` to runby python2.

gricn replied:
Ignore below, after referring to cerbot documentation I realize that time I used `--manual` but forgot to use `--prefer-challenges` so certbot to ask me certificate both http-01 and dns-01. (Ah, what a cheap mistake)

> Hold on, I test my script using the command `certbot renew --dry-run`. I remember that using certbot to add a wildcard domain now needs to:
> - add a `_acme-challenge` TXT record in the nameserver.
> - add a record in `http://example.com/.well-known/acme-challenge/xxxxxx_xxx-XXXXXX`.  
>
> This script only includes the first one,  so **it is not correct and needs to add a new part.**

alexzorin replied:
Good to hear. 

We could also detect the situation where an HTTP challenge is present, and remind the user to include `--preferred-challenges dns`. But, let's leave that for another time.

Also, with the `#!/usr/bin/env python` thing, my idea is to just leave it, and allow it to use to whatever version of Python is default on that system. `python` will automatically be a symlink to either `python2` or `python3`. I don't see any special need for the user to configure it themselves.

alexzorin reviewed (changes requested):
Hey, big thanks for contributing these changes.

Just a couple of things.

gricn committed:
Update vultr-dns.py

Edit as @alexzorin suggested:
- Replace `data` paramenter with `json` in `vultr_request` function
- Remove redundant "print"

gricn committed:
Update vultr-dns.py

Let users decide Python version

gricn commented:
I found that [acme.sh](https://github.com/acmesh-official/acme.sh) has already implemented this feature and has a better experience. It not only supports Letsencrypt but other certificate authorities such as Zerossl. In the spirit of not building wheels, I'd recommend just using this software.
